modules/cache_behavior/variables.tf

```diff
@@ -0,0 +1,135 @@
+variable "path_pattern" {
+  description = "The pattern (for example \"images/*.jpg\") that specifies which requests you want this cache behavior to apply to. Not required for the default cache behavior!"
+  type        = string
+  default     = null
+}
+
+variable "allowed_methods" {
+  description = "Controls which HTTP methods CloudFront processes and forwards to your Amazon S3 bucket."
+  type        = list(string)
+  default     = ["GET", "HEAD", "OPTIONS"]
+
+  validation {
+    condition = alltrue([
+      for v in var.allowed_methods : contains(["GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"], v)
```

Comment:
This is a nice validation 👍. I didn't know about `for` expressions in Terraform.